src/cabal

```diff
@@ -108,6 +109,23 @@ run_repl () {
 }
 
 #
+# Upgrade the script in place
+#
+run_upgrade () {
+    echo "Checking for a new version of cabal extras ..."
+    TZ=$(date +"%T")
+    curl --silent "https://raw.githubusercontent.com/ambiata/cabal-extras/master/src/cabal?$TZ" > /tmp/cabal
+    if ! cmp ./cabal /tmp/cabal  >/dev/null 2>&1
+    then
+      echo "New version found and upgraded. You can now commit it to your git repo."
+      mv /tmp/cabal ./cabal
```

Comment:
One thing we've wanted to do for a while is insert a hash comment with the version in github. /feature-request